spec/factories/theft_alerts.rb

```diff
@@ -0,0 +1,56 @@
+FactoryBot.define do
+  factory :theft_alert do
+    stolen_record { create(:stolen_record) }
+    theft_alert_plan { create(:theft_alert_plan) }
+    creator { create(:user_confirmed) }
+    status { "pending" }
+
+    trait :paid do
+      after(:create) do |theft_alert, _evaluator|
+        payment = create(:payment, user: theft_alert.creator)
+        theft_alert.update_attributes(payment: payment)
+      end
+    end
+
+    trait :begun do
+      status { "active" }
+      begin_at { Time.current.beginning_of_day }
+      sequence(:facebook_post_url) do |n|
+        "https://facebook.com/user.#{creator.id}/posts/#{n}"
+      end
+      before(:create) do |theft_alert, _evaluator|
+        plan = create(:theft_alert_plan)
```

Comment:
I'd prefer if, for both this an `ended`, you used a plan that would be assignable during factory creation (so that you can create multiple theft alerts referencing the same plan, etc)

I also don't believe you need to put this in a `before(:create)` block, I think you can just do:

```ruby
end_at { begin_at + theft_alert_plan.duration_days.days) } # optionally adding .end_of_day if necessary
```

Reply:
> I'd prefer if, for both this an ended, you used a plan that would be assignable during factory creation (so that you can create multiple theft alerts referencing the same plan, etc)

That's already the case -- it's a belongs_to

> I also don't believe you need to put this in a before(:create) block, I think you can just do:

That's not the case (the plan must be set for the record to be persisted), but if there's a workaround, let me know.